src/Mod/CAM/PathSimulator/AppGL/GuiDisplay.cpp

```diff
@@ -0,0 +1,453 @@
+/***************************************************************************
+ *   Copyright (c) 2024 Shai Seger <shaise at gmail>                       *
+ *                                                                         *
+ *   This file is part of the FreeCAD CAx development system.              *
+ *                                                                         *
+ *   This library is free software; you can redistribute it and/or         *
+ *   modify it under the terms of the GNU Library General Public           *
+ *   License as published by the Free Software Foundation; either          *
+ *   version 2 of the License, or (at your option) any later version.      *
+ *                                                                         *
+ *   This library  is distributed in the hope that it will be useful,      *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,         *
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "GuiDisplay.h"
+#include "OpenGlWrapper.h"
+#include "MillSimulation.h"
+#include <cstddef>
+#include "GlUtils.h"
+#include <stdlib.h>
+
+using namespace MillSim;
+
+static const char* ThumbImg =
```

Comment:
You are correct. I just do not know where to place the file.
Instead, if there is a way to embed a file as a resource, It would be nice. Do you know where can I find an example for that?

Reply:
I'll try to think of a good way - qt has resources capabilities so probably it would be good to render UI using QT instead of custom rendering in ogl